src/index.ts

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+
+import type {
+  Request as WorkerRequest,
+  ExecutionContext,
+} from '@cloudflare/workers-types/experimental';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+
+type Event = {
+  cron: string;
+  type: string;
+  scheduledTime: number;
+};
+type Env = Record<string, string>;
+
+class Logger {
+  #API_KEY;
+  #CHAT_ID;
+
+  constructor(env: Env) {
+    this.#API_KEY = env.API_KEY;
+    this.#CHAT_ID = env.TELEGRAM_LOG_CHAT_ID;
+  }
+
+  async #log2telegram(msg: string, disable_notification = true) {
+    if (this.#CHAT_ID) {
+      const logUrl = `https://api.telegram.org/bot${
+        this.#API_KEY
+      }/sendMessage?chat_id=${
+        this.#CHAT_ID
+      }&parse_mode=HTML&text=${msg}&protect_content=true&disable_notification=${
+        disable_notification ? 'true' : 'false'
+      }`;
+      await fetch(logUrl);
+    }
+  }
+
+  async error(msg: string, request: WorkerRequest) {
+    await this.#log2telegram(
+      `<pre><b><code>ERROR LOG</code></b></pre>${msg}\nRequest:<pre><code class="language-json">${JSON.stringify(
+        request,
+        null,
+        2,
+      )}</code></pre>`,
+      false,
+    );
+  }
+
+  async info(msg: string) {
+    await this.#log2telegram(msg);
+  }
+}
+
+export default {
+  async scheduled(event: Event, env: Env, ctx: ExecutionContext) {
+    // const logger = new Logger(env);
+    const now = new Date().toLocaleString('en-US', {
+      timeZone: 'Asia/Hong_Kong',
+    });
+    const photos = {
+      'Victoria Peak (looking towards the north-northeast)':
+        'https://www.hko.gov.hk/wxinfo/aws/hko_mica/vpb/latest_HD_VPB.jpg',
+      'Central (Victoria Harbour)':
+        'https://www.hko.gov.hk/wxinfo/aws/hko_mica/cp1/latest_HD_CP1.jpg',
+      'International Commerce Centre (looking towards the southeast)':
+        'https://www.hko.gov.hk/wxinfo/aws/hko_mica/ic1/latest_HD_IC1.jpg',
+      'International Commerce Centre (looking towards the southwest)':
+        'https://www.hko.gov.hk/wxinfo/aws/hko_mica/ic2/latest_HD_IC2.jpg',
+      // 'Tsim Sha Tsui (looking towards the west)': `https://www.hko.gov.hk/wxinfo/aws/hko_mica/hk2/latest_HD_HK2.jpg`,
+      // 'Sai Wan Ho (looking towards the east)': `https://www.hko.gov.hk/wxinfo/aws/hko_mica/swh/latest_HD_SWH.jpg`,
+    };
+    const inputMediaPhotos = _.map(photos, (url: string, location: string) => ({
+      type: 'photo',
+      media: `${url}?v=${event.scheduledTime}`,
+      caption: `${location}\n<pre>${now}</pre>`,
+      parse_mode: 'HTML',
+    }));
+    const url = `https://api.telegram.org/bot${
+      env.API_KEY
+    }/sendMediaGroup?chat_id=${
+      env.TELEGRAM_CRON_JOB_CHAT_ID
+    }&media=${JSON.stringify(inputMediaPhotos)}&disable_notification=true`;
+    console.log(222, url);
+    const result = await fetch(url);
+    console.log(333, result);
+    // await logger.info(`Scheduled job from <code>${event.cron}</code> at ${now}.\n${photo}`);
```

Comment:
So we can move these to `scheduledEventHandler.ts`?

Reply:
Done